File: ford-2.0-app-dev/ford-2.0-app-dev/src/model/form/NewsletterForm.ts
import { ContactPreferenceEnum, DocumentTypeEnum } from "../../utils/constants";

export interface NewsletterForm {
    id?: number;
    eventId?: number;
    eventName?: string;
    eventCode?: string;
    vehicleId?: number;
    vehicleName?: string;
    firstname?: string;
    lastname?: string;
    documentType?: DocumentTypeEnum;
    documentNumber?: string;
    email?: string;
    phoneArea?: string;
    phone?: string;
    contactPreference?: ContactPreferenceEnum;
    receiveInformation?: boolean;
    acceptConditions?: boolean;
    createdOn?: Date;
    modifiedOn?: Date;
    isSynchronized?: boolean;
    isSynchronizedWithSaleforce?: boolean;
    syncDate?: Date;
    syncFailed?: boolean;
}

export interface NewsletterFormToSync {
    origen: string | null;
    suborigen: string | null;
    campaing_id: string | null;
    modelo_desc: string | null;
    nombre: string | null;
    primer_apellido: string | null;
    tipo_doc: string | null;
    nro_doc: string | null;
    email: string | null;
    cod_area_tel: string | null;
    nro_tel: string | null;
    evento_id: string | null;
    evento_desc: string | null;
    canal_contacto_pref: string | null;
    optin_recibe_info: string | null;
    optin_terminos: string | null;
    submission_Date: string | null;
}

export interface NewsletterFormToSyncSaleForce {
    keys: { ID?: string },
    values: {
        EventID?: string;
        EventName?: string;
        Firstname?: string;
        Lastname?: string;
        Email?: string;
        PhoneNumber?: string;
        VehicleOfInterest?: string;
        RecieveInformation?: string;
        AcceptConditions?: string;
        FechaSincro?: string;
    }
}

File: ford-2.0-api-dev/ford-2.0-api-dev/FordEvents.Model/FordEventsDB/NewsletterForm.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace FordEvents.Model.FordEventsDB
{
    public partial class NewsletterForm
    {
        public long Id { get; set; }
        public long EventId { get; set; }
        public string Firstname { get; set; }
        public string Lastname { get; set; }
        public string Email { get; set; }
        public string VehicleOfInterest { get; set; }
        public bool RecieveInformation { get; set; }
        public bool AcceptConditions { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public DateTime? CreatedOn { get; set; }
        public Guid? ModifiedBy { get; set; }
        public Guid? CreatedBy { get; set; }
        public bool Deleted { get; set; }

        public virtual Event Event { get; set; }
    }
}
